### PR TITLE
bug/1847_invalid_chars_for_entity_type

### PR DESCRIPTION
**Pull request by kzangeli**
Fixed issue #1847 - check for invalid chars in attribute type.
Issue #1844 is also fixed by this modification.
Issue #1845 also.
And #1438.

*[ Note that the name of the branch is erroneous. ]*

### CODE REVIEW

File: src/lib/jsonParseV2/jsonRequestTreat.cpp [hunk 1]
```diff
@@ -46,7 +46,7 @@
 *
 * jsonRequestTreat - 
 */
-std::string jsonRequestTreat(ConnectionInfo* ciP, ParseData* parseDataP, RequestType requestType, JsonDelayedRelease* releaseP)
+std::string jsonRequestTreat(ConnectionInfo* ciP, ParseData* parseDataP, RequestType requestType, JsonDelayedRelease* releaseP, std::vector<std::string>& compV)
```
Comment (fgalan) [inline src/lib/jsonParseV2/jsonRequestTreat.cpp]: This is a long line, maybe the style

```
std::string jsonReqeustTreat
(
   ...
)
```

should be used.

**Comment by fgalan** [comment]
> CNR must be updated ...

The best moment will be once Orion 0.28.0 gets released and this PR branch synced with develop, given that the CNR will be flushed during that process.

File: test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Create entity E with attr A1, attr-type 'house:flat'
+# 02. Try to change A1 type to invalid 'house<flat>', with PUT /v2/entities/E/attrs/A1, see it fail
+# 03. Make sure step 2 DID NOT modify E/A1
+# 04. Try to change A1 type to non-plain-ascii 'habitacion', with PUT /v2/entities/E/attrs/A1, see it fail
```
Comment (fgalan) [inline test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test]: 'habitacion' or 'habitación' ?

(The actual test step is ok, I'm meaning just the comment and the echo msg...)

File: test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+# 05. Make sure step 4 DID NOT modify E/A1
+# 06. Try to change A1 type to a string that is longer than 256 chars
```
Comment (fgalan) [inline test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test]: + ", see it fail" (also in corresponding echo msg)

(To be homogeneous with step 2 and step 4)

File: test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+# 07. Make sure step 5 DID NOT modify E/A1
+# 08. Try to change A1, adding a metadata with empty name
```
Comment (fgalan) [inline test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test]: + ", see it fail" (also in corresponding echo msg)

(To be homogeneous with step 2 and step 4)

File: test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+# 09. Make sure step 7 DID NOT modify E/A1
+#
+
+echo "01. Create entity E with attr A1, attr-type 'house<flat>'"
```
Comment (fgalan) [inline test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test]: But attribute type is "house:flat"... maybe the echo message (and corresponding comment in the .test preamble) is wrong?

Comment (kzangeli) [inline test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test]: Ah, yes, of course.

File: test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test (hunk 2 above)
Comment (kzangeli) [inline test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test]: I don't like very much to use these 'funny' chars. They always gives problems.
The test is with, where it is mandatory, in the comment it is not (non-plain-ascii describes it)

Comment (fgalan) [inline test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test]: Ok, NTC.

File: src/lib/jsonParseV2/jsonRequestTreat.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/jsonParseV2/jsonRequestTreat.cpp]: I did it in the header file (the line is 8 chars longer :-)) so yes, I will do it in the source file as well.

File: test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test (hunk 5 above)
Comment (kzangeli) [inline test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test]: Fixed in 6fc5d65

File: test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test (hunk 4 above)
Comment (kzangeli) [inline test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test]: Fixed in 6fc5d65

File: test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test (hunk 3 above)
Comment (kzangeli) [inline test/functionalTest/cases/1847_invalid_chars_for_attribute_type/invalid_chars_for_attribute_type.test]: Fixed in 6fc5d65

File: src/lib/jsonParseV2/jsonRequestTreat.cpp (hunk 1 above)
Comment (kzangeli) [inline src/lib/jsonParseV2/jsonRequestTreat.cpp]: Fixed in 6fc5d65

**Comment by fgalan** [comment]
Except for the CNR entry missing (we need to wait until Monday for that), LGTM.

**Comment by kzangeli** [comment]
CHANGES_NEXT_RELEASE fixed. PR ready.

**Comment by crbrox** [comment]
LGTM

**Pull request merged**